boa_engine/src/context/intrinsics.rs

```diff
@@ -0,0 +1,368 @@
+use crate::{
+    builtins::{error::r#type::create_throw_type_error, iterable::IteratorPrototypes},
+    object::{JsObject, ObjectData},
+    Context,
+};
+
+#[derive(Debug, Default)]
+pub struct Intrinsics {
+    /// Cached standard constructors
+    pub(super) standard_constructors: StandardConstructors,
+    /// Cached intrinsic objects
+    pub(super) objects: IntrinsicObjects,
+}
+
+impl Intrinsics {
+    /// Return the cached intrinsic objects.
+    #[inline]
+    pub fn objects(&self) -> &IntrinsicObjects {
+        &self.objects
+    }
+
+    /// Return the core standard objects.
+    #[inline]
+    pub fn standard_constructors(&self) -> &StandardConstructors {
```

Comment:
```suggestion
    pub fn constructors(&self) -> &Constructors {
```
Since we have to call `.intrinsics()` before we can get the standard constructors, I think it's a bit redundant to call it `standard_constructors()` now.
```rust
context
    .intrinsics()
    .constructos()
    .type_error()
    .prototype()
```

Reply:
Fixed